distiller/data/models/pdf_extract.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from django.db import models
+from django.contrib.postgres.fields import JSONField
+
+from compositefk.fields import CompositeForeignKey
+
+from .single_audit_db import Audit
+
+
+class PDFExtract(models.Model):
+    """
+    PDF extract, per audit number.
+    Attempt to extract as much data from the PDF as we can.
+    """
+    class Meta:
+        verbose_name_plural = 'PDF extracts'
+
+    # Use these fields to link tables- 4 digits
+    audit_year = models.DecimalField(
+        max_digits=4,
+        decimal_places=0,
+        help_text='Audit Year and DBKEY (database key) combined make up the primary key.'
+    )
+    # Use these fields to link tables- 1-6 digits
+    dbkey = models.CharField(
+        max_length=6,
+        help_text='Audit Year and DBKEY (database key) combined make up the primary key.'
+    )
+    # Map to General/Audit
+    audit = CompositeForeignKey(
+        Audit,
+        on_delete=models.DO_NOTHING,
+        to_fields={
+           'audit_year': 'audit_year',
+           'dbkey': 'dbkey'
+        },
+        related_name='pdf_extracts'
+    )
```

Comment:
An index on `(audit_year, dbkey)` should help speed this join up.

Reply:
Absolutely right, thanks for the catch. Will add this.